src/mongoose_traffic_channel.erl

```diff
@@ -0,0 +1,241 @@
+%%%===================================================================
+%%% @copyright (C) 2016, Erlang Solutions Ltd.
+%%% @doc Module providing support for websockets in MongooseIM
+%%% @end
+%%%===================================================================
+-module(mongoose_traffic_channel).
+
+-behaviour(cowboy_websocket).
+
+%% cowboy_http_websocket_handler callbacks
+-export([init/2,
+         websocket_init/1,
+         websocket_handle/2,
+         websocket_info/2,
+         terminate/3]).
+
+-include("mongoose.hrl").
+-include("jlib.hrl").
+
+-define(LISTENER, ?MODULE).
+-define(MAX_ITEMS, 500).
+-define(MAX_ACCOUNTS, 100).
+
+-record(state, {traces = #{},
+                tracing = false,
+                current = <<>>,
```

Comment:
What's the use of `current`? Difficult to tell after a quick read.

Reply:
This is pid you are currently following. The monitor is accumulating stanzas from all traces pids, but one of them is the current one (you have it on your screen) and its stanzas are being streamed to you.